8293304: Replace some usages of INTPTR_FORMAT with PTR_FORMAT

stefank opened the pull request:
During the discussion of [JDK-8292981](https://bugs.openjdk.org/browse/JDK-8292981) an opinion was voiced that we should stop using INTPTR_FORMAT when printing pointers.

Some background that could explain why some tend to use INTPTR_FORMAT instead of PTR_FORMAT:

Both those format specifiers require an integer and the compiler barfs if you send in a pointer. We therefore have a utility function named p2i, which converts the pointer to an integer. So, everywhere we have to write print line like this: `print("my pointer: " PTR_FORMAT, p2i(my_pointer));`. Now, p2i returns an intptr_t, and it becomes natural for some to consider the type of the converted value (that we need because of the mentioned workaround), instead of the original type of the value.

With this enhancement I'd like to clean up the code that I often work in, so that it uses PTR_FORMAT when printing pointers. I'm leaving the rest of the code base for others to consider cleaning up.

Cleanups have been done in directories gc/, utilities/, memory/, and oops/.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293304](https://bugs.openjdk.org/browse/JDK-8293304): Replace some usages of INTPTR_FORMAT with PTR_FORMAT


### Reviewers
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to [e3d8f643](https://git.openjdk.org/jdk/pull/10141/files/e3d8f64320f99d4f98500ae6dbcedb827d06bc18)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10141/head:pull/10141` \
`$ git checkout pull/10141`

Update a local copy of the PR: \
`$ git checkout pull/10141` \
`$ git pull https://git.openjdk.org/jdk pull/10141/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10141`

View PR using the GUI difftool: \
`$ git pr show -t 10141`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10141.diff">https://git.openjdk.org/jdk/pull/10141.diff</a>

</details>

stefank commented:
@kimbarrett You commented on the "incorrect" usages of INTPTR_FORMAT in JDK-8292981. Do you think this PR moves the code in the right direction?

kimbarrett reviewed (approved):
Looks good.

I did notice a couple INTPTR_FORMAT => (new) SIZE_FORMAT_X_0.  Good change.

I also noticed a place where there are sequential asserts, one (previously)
using INTPTR_FORMAT and the next PTR_FORMAT, for the same value. Seems like
somebody wasn't paying attention. Again good change.

Also noticed some pre-existing unnecessary (void*) casts of the pointer
argument to p2i. Not your problem, and not really appropriate for this change
to do anything about them, but I really hate eye-catching cast junk like that.

There were also a few INTPTR_FORMAT uses that were left in place. All that I
recall looked like they could/should instead be SIZE_FORMAT_X_0 or maybe even
SIZE_FORMAT_X if the zero padding isn't useful. Leaving those for a later pass
seems fine. I wonder how many INTPTR_FORMATs are left in the areas you worked
on?

stefank commented:
> Looks good.

Thanks.
 
> Also noticed some pre-existing unnecessary (void*) casts of the pointer argument to p2i. Not your problem, and not really appropriate for this change to do anything about them, but I really hate eye-catching cast junk like that.

I agree. I cleaned out some of that. There's a lot of casts from `oop` to `void*` that are unnecessary after we rewrote the fastdebug oop class and/or `p2i`.

> 
> There were also a few INTPTR_FORMAT uses that were left in place. All that I recall looked like they could/should instead be SIZE_FORMAT_X_0 or maybe even SIZE_FORMAT_X if the zero padding isn't useful. Leaving those for a later pass seems fine. I wonder how many INTPTR_FORMATs are left in the areas you worked on?

I saw a lot of intptr_t/uintptr_t being printed with INTPRT_FORMAT, I don't think those should be changed to use SIZE_FORMAT_.

I found and converted three more lingering INTPTR_FORMAT usages. We are now down to a little bit less then 30 usages of INTPTR_FORMAT in the directories relevant to this PR. Most in:
g1HeapRegionAttr.hpp to print an index
elfFile.hpp to print a integer typed address
markWord.cpp to print the mark word
stackChunk/instanceStackChunk files to print relative stack pointers
